include/net/cloud.h

```diff
@@ -109,6 +109,21 @@ struct cloud_event {
 	} data;
 };
 
+/**
+ * @brief Cloud last will and testament type. See:
+ * http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html
```

Comment:
Here we introduce MQTT-specific stuff into a theoretically generic API. @jtguggedal FYI, as you're the author of it.

Probably we should encapsulate cloud-specific paramters in a more generic structure, which could be extended as more clouds are supported in the future. Something like:
```
struct cloud_cfg {
	union {
		struct {
			u8_t qos;
			bool retain;
			const char *topic;
			const char *message;
		} aws;

		struct {
			...
		} some_other_cloud_we_might_support;
	}
};
```

Reply:
Yeah, sounds good to keep it more generic when possible

Reply:
This isn't an AWS-specific feature, it's an MQTT-specific feature, so I would not want to name the struct inside the union 'aws'.  And the other struct would be for some_other_transport I suppose (different from MQTT).